client/components/domains/transfer-domain-step/index.jsx

```diff
@@ -0,0 +1,260 @@
+									" all in one place. Domains purchased in the last 60 days can't be transferred. {{a}}Learn More{{/a}}",
+								{
+									components: { a: <a href="#" /> },
+								}
+							) }
+						</p>
+					</div>
+
+					<div className="transfer-domain-step__add-domain" role="group">
+						<input
+							className="transfer-domain-step__external-domain"
+							type="text"
+							value={ this.state.searchQuery }
+							placeholder={ translate( 'example.com' ) }
+							onBlur={ this.save }
+							onChange={ this.setSearchQuery }
+							onClick={ this.recordInputFocus }
+							autoFocus
+						/>
+					</div>
+					<button
+						className="transfer-domain-step__go button is-primary"
+						onClick={ this.recordGoButtonClick }
+					>
+						{ translate( 'Transfer to WordPress.com' ) }
+					</button>
+					{ this.domainRegistrationUpsell() }
+					<div>
+						<p>
+							{ translate(
+								"Don't want to transfer? You can {{a}}map it{{/a}} for %(cost)s instead.",
+								{
+									args: { cost },
+									components: { a: <a href="#" onClick={ this.goToMapDomainStep } /> },
+								}
+							) }
+							<Gridicon icon="help" size={ 24 } />
+						</p>
+					</div>
+				</form>
+			</div>
+		);
+	}
+
+	domainRegistrationUpsell() {
+		const { suggestion } = this.state;
+		if ( ! suggestion ) {
+			return;
+		}
+
+		return (
+			<div className={ 'transfer-domain-step__domain-availability' }>
+				<Notice status="is-success" showDismiss={ false }>
+					{ this.props.translate( '%(domain)s is available!', {
+						args: { domain: suggestion.domain_name },
+					} ) }
+				</Notice>
+				<DomainRegistrationSuggestion
+					suggestion={ suggestion }
+					selectedSite={ this.props.selectedSite }
+					domainsWithPlansOnly={ this.props.domainsWithPlansOnly }
+					key={ suggestion.domain_name }
+					cart={ this.props.cart }
+					onButtonClick={ this.registerSuggestedDomain }
+				/>
+			</div>
+		);
+	}
+
+	registerSuggestedDomain = () => {
+		this.props.recordAddDomainButtonClickInMapDomain(
+			this.state.suggestion.domain_name,
+			this.props.analyticsSection
+		);
+
+		return this.props.onRegisterDomain( this.state.suggestion );
+	};
+
+	recordInputFocus = () => {
+		this.props.recordInputFocusInMapDomain( this.state.searchQuery );
+	};
+
+	recordGoButtonClick = () => {
+		this.props.recordGoButtonClickInMapDomain(
+			this.state.searchQuery,
+			this.props.analyticsSection
+		);
+	};
+
+	setSearchQuery = event => {
+		this.setState( { searchQuery: event.target.value } );
+	};
+
+	handleFormSubmit = event => {
+		event.preventDefault();
+
+		const domain = getFixedDomainSearch( this.state.searchQuery );
+		this.props.recordFormSubmitInMapDomain( this.state.searchQuery );
+		this.setState( { suggestion: null, notice: null } );
+
+		checkDomainAvailability( domain, ( error, result ) => {
+			const status = result && result.status ? result.status : error;
+			switch ( status ) {
+				case domainAvailability.MAPPABLE:
```

Comment:
Also, just a note that subdomains are also mappable... but not transferable ;)

Reply:
Have plans to add another param here. :)